listenbrainz/webserver/decorators.py

```diff
@@ -12,8 +12,12 @@ def crossdomain(origin='*', methods=None, headers=None,
     # Based on snippet by Armin Ronacher located at http://flask.pocoo.org/snippets/56/.
     if methods is not None:
         methods = ', '.join(sorted(x.upper() for x in methods))
+    default_headers = ["AUTHORIZATION", "CONTENT-TYPE"]
     if headers is not None and not isinstance(headers, string_types):
-        headers = ', '.join(x.upper() for x in headers)
+        for x in headers:
+            if x.upper() not in default_headers:
+                headers += ', '.join(x.upper())
```

Comment:
`join` takes an iterable not str as input. If str is given as input then it'll instead split characters and join those. So `", ".join("ABC")` will become `"A, B, C"` which is incorrect.

Reply:
doh!